plugins/modules/volume.py

```diff
@@ -175,11 +186,48 @@ def poll_func() -> bool:
                 "failed to wait for volume status: timeout period expired"
             )
 
-    def _wait_for_volume_active(self) -> None:
+    def _wait_for_volume_active(self, volume: Volume) -> None:
         self._wait_for_volume_status(
-            self._volume, {"active"}, self._timeout_ctx.seconds_remaining
+            volume, {"active"}, self._timeout_ctx.seconds_remaining
+        )
+
+    def _clone_volume(self) -> Volume:
+        params = self.module.params
+
+        source_id = params.get("source_volume_id")
+        source_volume = Volume(self.client, source_id)
+        source_volume._api_get()  # Force lazy-loading
+
+        # Check if source volume was found
+        if source_volume is None:
+            return self.fail(
+                msg="Volume with is {0} could not be found.".format(source_id)
+            )
+
+        # If regions don't match up, it is an invalid clone operation
+        if params.get("region") != source_volume.region.id:
+            return self.fail(
+                msg="Specified region does not match source volume region."
+            )
+
+        # Perform the clone operation
+        vol = request_retry(
+            lambda: self.client.post(
+                "/volumes/{}/clone".format(source_id),
+                data={"label": params.get("label")},
+            )
         )
 
+        cloned_volume = Volume(self.client, vol.get("id"))
+        cloned_volume._api_get()  # Force lazy-loading
+
+        self.register_action(
+            "Cloned volume with source id {0}".format(source_id)
+        )
+        self._wait_for_volume_active(cloned_volume)
```

Comment:
Can we let line 252 call `_wait_for_volume_active`, after the volume is assigned to `self._volume`, similar to what `_create_volume` handles it?

Reply:
The way `_create_volume` is currently written, I cannot call `_wait_for_volume_active` there because the volume is created in a one-liner return statement. I can modify the code so that it spans multiple lines but I think it would make the function unnecessarily bloated. What do you think?

Reply:
Took care of this!